client/src/app/shared/components/chip/chip.component.scss

```diff
@@ -0,0 +1,59 @@
+/**
+ * For further information, see https://css-tricks.com/switch-font-color-for-different-backgrounds-with-css/
+ */
+:host {
+    --os-chip-red: 0;
+    --os-chip-green: 0;
+    --os-chip-blue: 0;
+
+    /**
+     * The threshold at which colors are considered "light".
+     * Range: decimals from 0 to 1,
+     * recommended 0.5 - 0.6
+     */
+    --os-chip-threshold: 0.5;
+    /**
+     * The threshold at which a darker border will be applied.
+     * Range: decimals from 0 to 1,
+     * recommended 0.8+
+     */
+    --os-chip-border-threshold: 0.8;
+}
+
+.os-chip {
+    background: rgb(var(--os-chip-red), var(--os-chip-green), var(--os-chip-blue));
+
+    /**
+     * Calc using the
+     * W3C luma method
+     * lightness = (red * 0.299 + green * 0.587 + blue * 0.114) / 255
+     */
+    --red: calc(var(--os-chip-red) * 0.299);
+    --green: calc(var(--os-chip-green) * 0.587);
+    --blue: calc(var(--os-chip-blue) * 0.114);
+    --sum: calc(var(--red) + var(--green) + var(--blue));
+
+    --perceived-lightness: calc(var(--sum) / 255);
+
+    /**
+     * 1) Any lightness value above the threshold will be considered "light", therefore apply a black text color. Any bellow will be considered dark, and use white color.
+     * This results from appying either a sub-zero (negative) or a higher-than-100 lightness value, which are capped to 0 and 100 respectively, to a HSL declaration
+     */
+    color: hsl(0, 0%, calc((var(--perceived-lightness) - var(--os-chip-threshold)) * -10000000%));
+
+    /**
+     * 2) sets the border as a 50% darker shade of the base color, ONLY if background color luma is higher than the border threshold.
+     * To achieve this I use the same sub-zero or higher-than-max technique, only this time using the Alpha value from an RGBA declaration.
+     * This results in a border that's either fully transparent or fully opaque
+     */
+    --border-alpha: calc((var(--perceived-lightness) - var(--os-chip-border-threshold)) * 100);
+
+    border-width: 0.2em;
+    border-style: solid;
+    border-color: rgba(
```

Comment:
I do not like having a different border color for bright colors. Please just use the background color. All in all I did only see the tag in the dialog due to autoupdate issues: For me they are a bit too large. They should have the same appearance as the tags used in OS3. Can you equalize the style?

-> This is the same point I meantioned above: Maybe you can use this component for both of them allowing various inputs.